multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/jobs/OrphanedFilesCleaner.java

```diff
@@ -0,0 +1,88 @@
+package org.cloudfoundry.multiapps.controller.process.jobs;
+
+import org.cloudfoundry.multiapps.common.SLException;
+import org.cloudfoundry.multiapps.controller.persistence.model.FileEntry;
+import org.cloudfoundry.multiapps.controller.persistence.services.FileService;
+import org.cloudfoundry.multiapps.controller.persistence.services.FileStorageException;
+import org.cloudfoundry.multiapps.controller.persistence.services.OperationService;
+import org.cloudfoundry.multiapps.controller.process.Messages;
+import org.cloudfoundry.multiapps.controller.process.flowable.FlowableFacade;
+import org.cloudfoundry.multiapps.controller.process.variables.Variables;
+import org.flowable.variable.api.history.HistoricVariableInstance;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.text.MessageFormat;
+import java.time.LocalDateTime;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@Named
+public class OrphanedFilesCleaner {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OrphanedFilesCleaner.class);
+
+    private final FileService fileService;
+    private final OperationService operationService;
+    private final FlowableFacade flowableFacade;
+
+    @Inject
+    public OrphanedFilesCleaner(FileService fileService, OperationService operationService,
+                                FlowableFacade flowableFacade) {
+        this.fileService = fileService;
+        this.operationService = operationService;
+        this.flowableFacade = flowableFacade;
+    }
+
+    //this is quite inefficient because the Operation table does not contain any info on which archive
+    //it is started with
+    @Scheduled(fixedRate = 30, timeUnit = TimeUnit.MINUTES)
```

Comment:
Why does it run every 30 minutes, when it gets started operations for one hour ago?

Reply:
My reasoning for the job execution being more often than the cleanup window is that if one file is left without an operation right after execution, it will be cleaned up sooner, rather than wait the whole hour. This will minimize the risk of a file-based DoS attack.